viewing_party/party.py

```diff
@@ -35,40 +33,31 @@ def watch_movie(user_data, title):
 # ------------- WAVE 2 --------------------
 
 def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0
     sum = 0
-    average = 0
+    for data in user_data["watched"]:
+        sum += data["rating"]
+    return sum / len(user_data["watched"])
 
-    for i in range(len(user_data["watched"])):
-        sum += user_data["watched"][i]["rating"]
-        average = sum  / len(user_data["watched"])
-
-    return average
 
 def get_most_watched_genre(user_data):
-
-    index = 0
-    genres = []
-    least_popular = []
-    most_popular = []
-    
-    if user_data == {"watched":[]}:
+    if not user_data["watched"]:
         return None
-    else:
-        for movie in user_data["watched"]:
-            index +=1
-            genres.append(user_data["watched"][index-1]["genre"])
-        for item in genres:
-            if item not in least_popular:
-                least_popular.append(item)
-            else:
-                most_popular.append(item)
-        return (most_popular[0])
+    most_watched = {}
+    for data in user_data["watched"]:
+        if data["genre"] not in most_watched:
+            most_watched[data["genre"]] = 1 
+        else:
+            most_watched[data["genre"]] += 1
+    the_max = max(most_watched, key = most_watched.get)
+    return the_max

```

Comment:
Really nice creation of a frequency dictionary! And really nice use of `max`! Can you also consider how you would find the most watched movie without using `max`? Which way do you personally prefer?